core/src/test/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManagerTests.java

```diff
@@ -171,4 +172,90 @@ public void authenticateWhenPostAuthenticationChecksNotSet() {
 
 		verifyZeroInteractions(this.postAuthenticationChecks);
 	}
+
+	@Test
+	public void testAuthenticateFailsIfAccountExpired() {
+		this.manager.setPasswordEncoder(this.encoder);
+		this.manager.setUserDetailsService(new MockUserDetailsServiceUserAccountExpired());
+
+		UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(
+				this.user, this.user.getPassword());
+		try {
```

Comment:
It would be nice to have the exception declared in the annotation `@Test(expected = LockedException.class)`, instead of using the try-fail-catch pattern. In other tests we catch the exception only if we need to validate an attribute, such as the message.